File: Common/Data/src/main/java/com/greentree/data/resource/location/NamedResourceLocation.java
package com.greentree.data.resource.location;

import java.util.HashMap;
import java.util.Map;

import com.greentree.data.resource.Resource;

public interface NamedResourceLocation extends ResourceLocation {

	Iterable<String> getAllNames();

	default Map<String, Resource> toMap() {
		final var map = new HashMap<String, Resource>();
		for(var name : getAllNames()) map.put(name, getResource(name));
		return map;
	}

}

File: Commons/Data/src/main/java/com/greentree/commons/data/resource/location/NamedResourceLocation.kt
package com.greentree.commons.data.resource.location

import com.greentree.commons.data.resource.Resource

/**
 * Интерфейс именованного местоположения ресурса.
 * Позволяет получать список всех имен ресурсов и преобразовывать
 * местоположение в Map.
 * 
 * @see ResourceLocation
 * @see MapResourceLocation
 */
interface NamedResourceLocation : ResourceLocation {

	/**
	 * Возвращает итератор по именам всех ресурсов.
	 */
	val names: Iterable<String>

	/**
	 * Преобразует местоположение в Map имя -> ресурс.
	 * @return Map имен ресурсов на сами ресурсы
	 */
	fun toMap(): Map<String, Resource> {
		val map = HashMap<String, Resource>()
		for(name in names)
			map[name] = getResource(name)
		return map
	}
}